biogeochem/EDPatchDynamicsMod.F90

```diff
@@ -1005,12 +1094,41 @@ subroutine spawn_patches( currentSite, bc_in)
        
       if ( site_areadis_primary .gt. nearzero) then
           currentPatch               => currentSite%youngest_patch
-          new_patch_primary%older    => currentPatch
-          new_patch_primary%younger  => null()
-          currentPatch%younger       => new_patch_primary
-          currentSite%youngest_patch => new_patch_primary
+          ! insert first primary patch after all the secondary patches, if there are any
+          if (currentPatch%anthro_disturbance_label .eq. secondaryforest ) then
+             foundfirstprimary = .false.
+             do while(associated(currentPatch) .and. .not. foundfirstprimary) 
+                currentPatch => currentPatch%older
+                if (associated(currentPatch)) then
+                   if (currentPatch%anthro_disturbance_label .eq. primaryforest) then
+                      foundfirstprimary = .true.
+                   endif
+                endif
+             end do
+             if (associated(currentPatch)) then
+                ! the case where we've found a youngest primary patch
+                new_patch_primary%older    => currentPatch
+                new_patch_primary%younger  => currentPatch%younger
+                currentPatch%younger%older => new_patch_primary
+                currentPatch%younger       => new_patch_primary
+             else
+                ! the case where we haven't, and are putting a primary patch at the oldest end of the 
+                ! linked list (not sure how this could happen, but who knows...)
```

Comment:
haha..

Reply:
actually, I think this can't happen in the current code, but if we use this as a template for the prescribed biography with competition then it might?  the way i think we could generalize this is to make an integer patch label that includes both the pft type (if used) and the anthro land use type and then iterate through this to put a given new patch such that they are in patch-label order and then within that in age order?

Reply:
actually, thinking about this even more, i think it could possibly happen if there were only one primary patch that was wholly consumed by natural disturbance in a single day.  not sure how that might happen but I'm sure in an ESM it somehow will.  so best to allow for that edge case.